cfgrammar/src/lib/yacc/ast.rs

```diff
@@ -17,6 +17,7 @@ pub struct GrammarAST {
     pub rules: IndexMap<String, Rule>,
     pub prods: Vec<Production>,
     pub tokens: IndexSet<String>,
+    pub spans: Vec<(usize, usize)>,
```

Comment:
This `(usize, usize`) type makes me wonder if we should put `Span` in `cfgrammar`, so that `lrpar` can access it, but `cfgrammar` can also use it. [Subtext: I think `Span` is clearer in intent than `(usize, usize)`.] Any thoughts?

Reply:
[Though we could do this in a second PR. It's not strictly related to this PR.]

Reply:
Sounds good to me.

Reply:
I think this was easy enough that it was worth getting done before inclusion.

I wasn't exactly sure where in cfgrammar, I ended up putting it in a `utils` module, because yacc didn't seem right if we ever get other formats besides yacc.  But open to thoughts here if you have other ideas...